lib/ansible/plugins/inventory/openstack.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2012, Marco Vito Moscaritolo <marco@agavee.com>
+# Copyright (c) 2013, Jesse Keating <jesse.keating@rackspace.com>
+# Copyright (c) 2015, Hewlett-Packard Development Company, L.P.
+# Copyright (c) 2016, Rackspace Australia
+# Copyright (c) 2017, Red Hat, Inc.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+'''
+DOCUMENTATION:
+    name: openstack
+    plugin_type: inventory
+    short_description: OpenStack inventory source
+    description:
+        - Get inventory hosts from OpenStack clouds
+        - Uses openstack.(yml|yaml) YAML configuration file to configure the inventory plugin
+        - Uses standard clouds.yaml YAML configuration file to configure cloud credentials
+    options:
+        show_all:
+            description: toggles showing all vms vs only those with a working IP
+            type: boolean
+            default: False
+        use_server_id:
```

Comment:
When you said "standardize on inventory_hostname" ... I'm not sure what that means in this context. This particular flag is "should the cloud server uuid be the 'hostname' in the inventory with a group created for the name of the cloud server always, or should that only be done if there are more than one cloud server with a particular name"

Oh -maybe you mean ... ok. nevermind. I think I grok.